src/services/item/repository.ts

```diff
@@ -242,15 +244,29 @@ export const ItemRepository = AppDataSource.getRepository(Item).extend({
       .getRawOne();
   },
 
-  async getOwn(memberId: string): Promise<Item[]> {
-    return this.createQueryBuilder('item')
+  async getOwn(
+    memberId: string,
+    searchArgs: {
+      name: string;
+    },
+    args: PaginationArgs,
+  ): Promise<Paginated<Item>> {
+    const pageLimit = args.limit || ITEMS_LIST_LIMIT;
+    const skip = (args.page - 1) * pageLimit;
+    const [data, totalCount] = await this.createQueryBuilder('item')
       .leftJoinAndSelect('item.creator', 'creator')
       .innerJoin('item_membership', 'im', 'im.item_path @> item.path')
       .where('creator.id = :id', { id: memberId })
       .andWhere('im.permission = :permission', { permission: PermissionLevel.Admin })
       .andWhere('nlevel(item.path) = 1')
+      .andWhere("LOWER(item.name) LIKE '%' || :name || '%'", {
+        name: searchArgs.name.toLowerCase().trim(),
+      })
       .orderBy('item.updatedAt', 'DESC')
-      .getMany();
+      .skip(skip)
+      .take(pageLimit)
```

Comment:
After a documentation review, it appears it's better to use `limit` and `offset`
https://orkhan.gitbook.io/typeorm/docs/select-query-builder#adding-offset-expression